Deliverable 1

vctrgtzm opened the pull request:
Codesandbox: https://codesandbox.io/s/boring-williams-5g4hd?file=/src/index.js
App deployed in Netlify: https://csb-bfs6j.netlify.app/

vctrgtzm committed:
Removed css file to use styled components globalStyle approach

vctrgtzm committed:
Add app component tests

vctrgtzm committed:
Fix test description

vctrgtzm committed:
Add global style an the top of react tree

vctrgtzm committed:
Make App component tests pass

vctrgtzm committed:
Add Header component test suite

vctrgtzm committed:
Added test item to Header component

vctrgtzm committed:
Remove unnecessary import and fixed syntax error

vctrgtzm committed:
Make Header component tests pass

vctrgtzm committed:
Import font and fix document identing

vctrgtzm committed:
Style header and its components

vctrgtzm committed:
Increase header height

vctrgtzm committed:
Add mock

vctrgtzm committed:
Add test id to theme toggle

vctrgtzm committed:
Create test suite for MainView component

vctrgtzm committed:
Fixed MainView test

vctrgtzm committed:
Make test pass

vctrgtzm committed:
Change background color when search input is on focus

vctrgtzm committed:
Removed non-working code

vctrgtzm committed:
Added background color to body

vctrgtzm committed:
Update App.jsx to use self-closing tags on the components rendered

vctrgtzm committed:
Add VideoGrid and VideoCard components and render them in MainView

vctrgtzm committed:
Make header responsive

vctrgtzm committed:
Test App to render a view in general, not the main view specifically

vctrgtzm committed:
Improve test queries by querying by role instead of by test id

vctrgtzm committed:
Remove create-react-app default readme

vctrgtzm commented:
I faced the following error when making commits using the fork of the [repo](https://github.com/cvillanueva84/react-certification-2021) provided in the [instructions of the deliverable 1](https://gist.github.com/cvillanueva84/fd19fa1d5e9feb6eab11bf732c483041), and to be honest I didn't have enough time to figure out why. So I created this repo to work on the project. Hope this is not an issue, if so, please let me know and I will try to work on the fork of the suggested [repo](https://github.com/cvillanueva84/react-certification-2021).

![CommitError](https://user-images.githubusercontent.com/49415800/127200003-93bf244d-a019-4a6c-b9ae-1ff1307a2184.JPG)

omarscd commented on src/globalStyle.js:
```diff
@@ -0,0 +1,11 @@
+    }
+`;
+
+export default GlobalStyle;
```
es una práctica común dejar una nueva línea al final del documento, aquí un link con algunas de las ventajas de hacerlo https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

vctrgtzm replied:
Es bueno saberlo, gracias! Nueva línea agregada al final de todos los archivos a lo que les faltaba en el nuevo commit a52aa80

omarscd commented on src/components/Header/Header.styled.js:
```diff
@@ -0,0 +1,92 @@
+
+export const HeaderSectionRight = styled(HeaderSection)`
+    justify-content: flex-end;
+`;
```
Si usas la propiedad `justify-content: space-between;` en `StyledHeader` te puedes evitar el uso de los 3 componentes de abajo.
https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content

vctrgtzm replied:
Tiene sentido. Sin embargo, al utilizar ese approach creo que es más complicado mantener centrado el `SeachInput` respecto al ancho del viewport. Estará centrado respecto al contenedor de los elementos del lado derecho e izquierdo, pero siempre se verá un poco más alineado a la derecha respecto al viewport debido a que el contenedor del lado izquierdo es más grande que el derecho. De esta manera:

![image](https://user-images.githubusercontent.com/49415800/127417724-7e4c9c0f-ded0-4462-ad0d-980804ed28e5.png)

Tiene sentido? O hay algo que no estoy viendo?

Por otro lado, los componentes `HeaderSectionCenter` y `HeaderSectionLeft` terminaron teniendo más propiedades con los cambios en el commit 890ff0c por lo que creo que podría ser otra buena razón para mantener estos componentes. Igual gradeceré tu opinión una vez sabiendo por qué lo hice así, o si hay algúna otra recomendación al respecto.

omarscd replied:
Tiene todo el sentido del mundo, no había considerado ese detallito.

omarscd commented on src/components/Header/Header.styled.js:
```diff
@@ -0,0 +1,92 @@
+    }
+`;
+
+export const NavList = styled.ul`
```
Para secciones con links de navegación puedes usar la etiqueta `<nav>` https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav

vctrgtzm replied:
Fue corregido en el proceso de refactorización en el commit 890ff0c :grimacing:

omarscd commented on src/views/MainView/MainView.jsx:
```diff
@@ -1,6 +1,14 @@
+    return <section data-testid="main-view">
+        <div>
+            {mockData.items.map((item, idx) => {
+                return <div key={idx} data-testid="video-container">{item.snippet.title}</div>
```
Se recomienda que el el `key` cuando se tiene una lista de varios elementos no sea el número de su index, sino un string que identifique de manera estable ese item.
https://reactjs.org/docs/lists-and-keys.html#keys

vctrgtzm replied:
Fue corregido en el proceso de refactorización en el commit 3a039bf :blush:

omarscd commented on src/components/Header/Header.styled.js:
```diff
@@ -65,7 +65,11 @@ export const SearchInput = styled.input.attrs({type: 'text'})`
-    width: 50%
+    width: 50%;
+
+    &:focus{
+        background-color: #e9ecef;
+    }
```
nice 💅🏼

vctrgtzm replied:
:see_no_evil:

omarscd commented on src/components/VideoCard/VideoCard.styled.js:
```diff
@@ -0,0 +1,44 @@
+overflow: hidden;
+background-color: #f8f9fa;
+font-family: 'Amatic SC', Arial;
+box-shadow: 0px 0px 20px 0px rgba(0,0,0,0.25);
+cursor: pointer;
+color: #03045E;
+transition: background-color .3s ease-out;
+
+&:hover {
+    background-color: #e9ecef;
+}
+
+&:hover > div {
+    background-size: 120%;
+}
+
+& > h2{
+    margin: 12px;
+    letter-spacing: 1px;
+}
+
+& > p {
+    font-size: 16px;
+    letter-spacing: 2px;
+    font-weight: bold;
+    margin: 12px;
+
+}
+`;
+
+export const ThumbnailContiner = styled.div`
+min-height: 185px;
+background-image: url(${props => props.thumbnail});
+background-size: 110%;
+background-repeat: no-repeat;
+background-position: center;
+
+transition: background-size .3s ease-out;
+`;
+
```
detalle pequeño en este documento, creo que faltó checar la indentación

vctrgtzm replied:
Indentación agregada en nuevo commit 4103d2f

omarscd commented on src/components/VideoGrid/VideoGrid.jsx:
```diff
@@ -0,0 +1,22 @@
+            return (
+                <VideoCard
+                    data-testid="video-card"
+                    key={item.etag}
```
perfecto

vctrgtzm replied:
:raised_hands:

omarscd commented on src/components/Header/Header.styled.js:
```diff
@@ -25,10 +29,21 @@ export const HeaderSectionLeft = styled(HeaderSection)`
 export const HeaderSectionRight = styled(HeaderSection)`
     justify-content: flex-end;
+    padding-left: 16px;
+    @media(max-width: 768px){
+        display: none;
+    }
```
alternativamente puedes crear una clase que contenga esta regla y agregársela a todos los elementos que quieras que se oculten en pantallas pequeñas para que no tengas que estar escribiendo esta regla una y otra vez en diferentes componentes, por ejemplo:
```
@media(max-width: 768px){
  .hidden-mobile {
    display: none;
  }
}
```

vctrgtzm replied:
Suena bien! Gracias por la recomendación. 😃 Implementado en commit  a49d16c

omarscd commented on src/components/Header/Header.styled.js:
```diff
@@ -41,35 +56,55 @@ export const Logo = styled.img`
+    &:hover{
+        color: #48cae4;
         cursor: pointer;
     }
```
también estaría bien agregarla un transition para que el cambio de color no se vea tan abrupto

vctrgtzm replied:
Cierto! Es más agradable con transition. Agregado en commit 37ed6c8 👍

omarscd commented on src/components/App/App.test.js:
```diff
@@ -7,12 +7,21 @@ describe('App', () => {
-          const view = screen.queryByTestId("view");
-          expect(view).toBeInTheDocument();
-      });
+    //TEST NOT WORKING WITH HEADER BY ROLE, TODO: FIND OUT WHY
+
+    // test('should render the header', () => {
+    //     render(<App />);
+
+    //     const headerComponent = screen.queryByRole("heading");
+    //     expect(headerComponent).toBeInTheDocument();
+    // });
```
recuerda que puedes usar la función `test.todo` para casos como este
https://dev.to/karltaylor/use-test-todo-when-writing-jest-tests-4dbd

vctrgtzm replied:
Revisaré cómo funciona, gracias!

vctrgtzm replied:
Interesante, es bueno saber que existe esta función. La utilizo en el commit bc55c63 pero me temo que aquí voy a necesitar tu ayuda, no he descubierto qué estoy haciendo mal o por qué me está dando el error `TestingLibraryElementError: Found multiple elements with the role "heading"` siendo que solamente existe un elemento con `role="heading"`. Alguna idea de qué puede estar pasando?

omarscd replied:
Lo elementos que tienen `role=heading` son `<h1>-<h6>`, que creo que tienes varias en la app, entonces me hace sentido que te arroje error. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role
Ahora, si lo que quieres es obtener el `<header>`, ese tiene un `role=banner`. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role
Un poco confuso, no?

vctrgtzm replied:
Tienes toda la razón! Siendo sincero no sabía que algunos elementos tienen un `role` implícito. De nuevo te agradezco por iluminarme 🤝

vctrgtzm replied:
Corregido en baf0415

omarscd reviewed (changes requested):
Bien hecho 👍🏼

vctrgtzm committed:
Added newline at the end of files missing it.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

vctrgtzm committed:
Indentation added

vctrgtzm committed:
Removed commented code

vctrgtzm committed:
Add classes to show or hide elements on mobile devices

vctrgtzm committed:
Added transition on hover navigation elements

vctrgtzm committed:
Use test.todo

vctrgtzm commented:
Muchas gracias por el feedback y los recursos de lectura! @omarscd

omarscd reviewed (approved):
LGTM

vctrgtzm committed:
Change <header> role to "banner" and test it is rendered querying by its role
